### ISSUE TITLE
enable MPI domain decomposition with plugins

### ISSUE THREAD

**Issue by eirrgang** [kassonlab/gmxapi#50]
Thread-MPI domain decomposition is working, but we should allow MPI domain decomposition with plugins. This requires some more fiddling with the MPI initialization in and out of GROMACS and sharing of MPI communicators.

**Comment by peterkasson** [comment]
Yes--as you know, this is high priority for some of our target applications.

**Comment by eirrgang** [comment]
This is kind of an involved thing to trouble-shoot and has caused a lot of headaches in switching back and forth between tool chains. I want to isolate it as much as possible and keep up to date with the changes occurring in the communicator interfaces upstream. I suggest that I pursue MPI communicator sharing in a fresh fork of GROMACS 2018 with minimal gmxapi features to start with. I would like to start more or less immediately so that the solution can be integrated with gmxapi before much more gmxapi development adds complexity.

**Comment by eirrgang** [comment]
Also ref https://github.com/kassonlab/sample_restraint/pull/15
